### PR TITLE
Cascading Delete for PlaylistFavorites if Playlist or Favorite is Deleted

### PR DESCRIPTION
**Pull request by fentontaylor**
### Bug Description
- A user could delete a playlist or a favorite if it was referenced in the playlist_favorites table.

### Fixed In This PR
- Added a new migration to add `ON DELETE CASCADE` so that when a `playlist` or a `favorite` is deleted, all records in the `playlist_favorites` table with the respective ids are also deleted
- Adds tests in ...playlists/destroy.spec.js and ...favorites/destroy.spec.js to ensure deletion works as expected.

### Also Done
- Revamped seed files to include a few extra records
- Checked column attributes in Postico to ensure that migration and rollback update the `on delete` action correctly and maintain all other desired attributes.

### CODE REVIEW

File: db/migrations/20191210194854_addCascadeOnDeleteToPlaylistFavorites.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+exports.up = function(knex) {
+  return knex.schema.table('playlist_favorites', (table) => {
+    table.dropForeign('playlist_id', 'playlist_favorites_playlist_id_foreign');
+    table.dropForeign('favorite_id', 'playlist_favorites_favorite_id_foreign');
+  })
+  .then(() => {
+    return knex.schema.alterTable('playlist_favorites', (table) => {
+      table.integer('playlist_id').unsigned()
```
Comment (nathangthomas) [inline db/migrations/20191210194854_addCascadeOnDeleteToPlaylistFavorites.js]: what functionality does .unsigned() provide?

File: db/migrations/20191210194854_addCascadeOnDeleteToPlaylistFavorites.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+
+exports.up = function(knex) {
+  return knex.schema.table('playlist_favorites', (table) => {
+    table.dropForeign('playlist_id', 'playlist_favorites_playlist_id_foreign');
```
Comment (nathangthomas) [inline db/migrations/20191210194854_addCascadeOnDeleteToPlaylistFavorites.js]: where does `playlist_favorites_playlist_id_foreign` come from?

Comment (fentontaylor) [reply db/migrations/20191210194854_addCascadeOnDeleteToPlaylistFavorites.js]: It is the name of the foreign key constraint, which by default is in the format table_name + column_name + foreign.

File: db/migrations/20191210194854_addCascadeOnDeleteToPlaylistFavorites.js (hunk 1 above)
Comment (fentontaylor) [reply db/migrations/20191210194854_addCascadeOnDeleteToPlaylistFavorites.js]: It is a SQL term used when creating an integer column that indicates it should only be positive, allowing for a greater upper range of value. Commonly used for IDs [more detail here](https://stackoverflow.com/questions/3895692/what-does-unsigned-in-mysql-mean-and-when-to-use-it#3895705)

**Pull request merged**